src/alr/alr-commands-build.adb

```diff
@@ -4,23 +4,36 @@ with Alire.Properties.Actions.Executor;
 with Alr.Root;
 with Alr.Spawn;
 with Alr.Platform;
-with Alr.Build_Env;
 
 with GNAT.OS_Lib;
 
 package body Alr.Commands.Build is
 
-   ----------------
-   -- Do_Compile --
-   ----------------
+   -------------
+   -- Execute --
+   -------------
 
-   function Do_Compile return Boolean is
+   overriding procedure Execute (Cmd : in out Command) is
+      pragma Unreferenced (Cmd);
+   begin
+      if not Execute (Export_Build_Env => True) then
+         Reportaise_Command_Failed ("Compilation failed.");
+      end if;
+   end Execute;
+
+   -------------
+   -- Execute --
+   -------------
+
+   function Execute (Export_Build_Env : Boolean) return Boolean is
    begin
       Requires_Full_Index;
 
       Requires_Valid_Session;
 
-      Alr.Build_Env.Export (Alr.Root.Current);
+      if Export_Build_Env then
```

Comment:
If would be nice to export the environment just when we need it, when spawning a command.

In python it is doable with `popen`, but I don't know any similar API for Ada: https://docs.python.org/3/library/subprocess.html#subprocess.Popen

Reply:
Ah, I see. That's too bad we don't have something portable like that in GNAT or GNATCOLL...